src/lib/Form/DataTransformer/ContentInfoTransformer.php

```diff
@@ -27,17 +30,49 @@ public function __construct(ContentService $contentService)
         $this->contentService = $contentService;
     }
 
+    /**
+     * Transforms a domain specific ContentInfo object into a Content's ID.
+     *
+     * @param null|ContentInfo $value
+     * @return mixed|null
+     * @throws TransformationFailedException if the given value is not a ContentInfo object
+     */
     public function transform($value)
     {
-        return null !== $value
-            ? $value->id
-            : null;
+        if (null === $value) {
+            return null;
+        }
+
+        if (!$value instanceof ContentInfo) {
+            throw new TransformationFailedException('Expected a ' . ContentInfo::class . ' object.');
+        }
+
+        return $value->id;
     }
 
-    public function reverseTransform($value)
+    /**
+     * Transforms a Content's ID integer into a domain specific ContentInfo object.
+     *
+     * @param mixed $value
+     * @return ContentInfo|null
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws TransformationFailedException if the given value is not an integer
+     *                                       or if the value can not be transformed
+     */
+    public function reverseTransform($value): ?ContentInfo
     {
-        return null !== $value && !empty($value)
-            ? $this->contentService->loadContentInfo($value)
-            : null;
+        if (empty($value)) {
+            return null;
+        }
+
+        if (!is_int($value)) {
+            throw new TransformationFailedException('Expected an integer.');
+        }
+
+        try {
+            return $this->contentService->loadContentInfo($value);
+        } catch (NotFoundException $e) {
+            throw new TransformationFailedException('Transformation failed. ' . $e->getMessage(), $e->getCode(), $e);
```

Comment:
Do not prefix message with `Transformation failed.`. Better to rethrow exception with original message.